### PR TITLE
[sparkle] - refactor: streamline LinkWrapper rendering logic

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the `LinkWrapper` component integration in navigation components to improve rendering logic. Instead of wrapping entire content blocks with `LinkWrapper`, it now wraps only the interactive elements within `NavigationListItem` and `TabsTrigger`. It is also now being used within the `Button` component.

## Risk

Low

## Deploy Plan

Publish `sparkle`

### CODE REVIEW

File: sparkle/src/components/Button.tsx [hunk 1]
```diff
@@ -158,23 +167,31 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
...
+      <LinkWrapper
+        href={href}
+        target={target}
+        rel={rel}
+        replace={replace}
+        shallow={shallow}
       >
-        {content}
-      </MetaButton>
+        <MetaButton
```
Comment (flvndvd) [inline sparkle/src/components/Button.tsx:177]: Wait I thought that the button was support href, directly?

File: sparkle/src/components/Tabs.tsx [hunk 2]
```diff
@@ -56,28 +56,24 @@ const TabsTrigger = React.forwardRef<
...
-            disabled={disabled}
-          />
+          <LinkWrapper
+            href={href}
+            target={target}
+            rel={rel}
+            replace={replace}
+            shallow={shallow}
+          >
+            <Button
```
Comment (flvndvd) [inline sparkle/src/components/Tabs.tsx:66]: Same here doesn't button handles `href` out-of-the-box?

File: sparkle/src/components/Button.tsx (hunk 1 above)
Comment (JulesBelveze) [reply sparkle/src/components/Button.tsx:177]: Not that I know of? See [docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLButtonElement)

Comment (flvndvd) [reply sparkle/src/components/Button.tsx:177]: I meant our own `Button` component.

Comment (JulesBelveze) [reply sparkle/src/components/Button.tsx:177]: Nop it unfortunately doesn't and I'm trying to bring it here. Tried to follow the shadcn pattern -> wrapping the link inside a button but it then breaks the Tabs. Any suggestion ?

**Review by flvndvd — APPROVED** [review]
LGTM

**Pull request merged**